### PR TITLE
Merging SuperVoxels using Convexity Criteria

### PR DESCRIPTION
**Pull request by iKrishneel**
The nodelet merges small clusters of supervoxels into high level object representation using convexity criteria.

### CODE REVIEW

**Comment by k-okada** [comment]
Please update your .travis by git submodule update 
Also better to squash your commit into one , since most of the commit is to just delete wrong commits
http://makandracards.com/makandra/527-squash-several-git-commits-into-a-single-commit

File: jsk_pcl_ros/include/jsk_pcl_ros/region_adjacency_graph.h [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+      typedef typename boost::property_map<
+         Graph, boost::edge_weight_t>::type EdgePropertyAccess;
+      typedef typename boost::property_traits<boost::property_map<
+         Graph, boost::edge_weight_t>::const_type>::value_type EdgeValue;
+      typedef typename boost::graph_traits<
+         Graph>::vertex_iterator VertexIterator;
+      typedef typename boost::graph_traits<
+         Graph>::vertex_descriptor VertexDescriptor;
+   
+      Graph graph;
```
Comment (garaemon) [inline jsk_pcl_ros/include/jsk_pcl_ros/region_adjacency_graph.h]: Add `_` suffix to instance variables

File: jsk_pcl_ros/include/jsk_pcl_ros/convex_connected_voxels.h [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+#include <pcl/segmentation/extract_clusters.h>
+#include <pcl/filters/extract_indices.h>
+#include <pcl/common/centroid.h>
+#include <pcl/common/impl/common.hpp>
+#include <pcl/registration/distances.h>
+
+// JSK Nodelet
+#include <jsk_topic_tools/diagnostic_nodelet.h>
+
+namespace jsk_pcl_ros {
```
Comment (garaemon) [inline jsk_pcl_ros/include/jsk_pcl_ros/convex_connected_voxels.h]: bracket after newline in `namespace`

File: jsk_pcl_ros/include/jsk_pcl_ros/convex_connected_voxels.h [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+   class ConvexConnectedVoxels: public jsk_topic_tools::DiagnosticNodelet {
```
Comment (garaemon) [inline jsk_pcl_ros/include/jsk_pcl_ros/convex_connected_voxels.h]: brakcet after newline in class declaration

**Comment by garaemon** [comment]
It's not necessary but I hope you to follow coding convention of jsk_recognition programs.
[BilateralFilter](https://github.com/jsk-ros-pkg/jsk_recognition/blob/master/jsk_pcl_ros/src/bilateral_filter_nodelet.cpp) is a good simple example

File: jsk_pcl_ros/include/jsk_pcl_ros/region_adjacency_graph.h [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+         bool = true);
+      int getCommonNeigbour(
+         const std::vector<int> &,
+         const std::vector<int> &);
+      void computeCloudClusterRPYHistogram(
+         const pcl::PointCloud<PointT>::Ptr,
+         const pcl::PointCloud<pcl::Normal>::Ptr,
+         cv::Mat &);
+
+    public:
```
Comment (garaemon) [inline jsk_pcl_ros/include/jsk_pcl_ros/region_adjacency_graph.h]: Do not have multiple public sections. Please make it once.

**Pull request closed without merging**